Build failure: supermodel

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. Build supermodel
2. Get error

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/294532371

### Relevant log output

```console
Running phase: unpackPhase
unpacking source archive /nix/store/gz1kblc3bgxwvhc7rafj1cx1saq891xj-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
build flags: SHELL=/nix/store/5rkx2slmya83anp81xdb12gbscn96k93-bash-5.2p37/bin/bash NET_BOARD=1
Creating directory     : bin
Creating directory     : obj
Generating dependencies: Src/CPU/PowerPC/PPCDisasm.cpp -> obj/PPCDisasm.d
Compiling              : Src/CPU/PowerPC/PPCDisasm.cpp -> obj/PPCDisasm.o
Generating dependencies: Src/BlockFile.cpp -> obj/BlockFile.d
Compiling              : Src/BlockFile.cpp -> obj/BlockFile.o
Src/BlockFile.cpp: In member function 'void CBlockFile::ReadString(std::string*, uint32_t)':
Src/BlockFile.cpp:51:10: warning: ignoring return value of 'size_t fread(void*, size_t, size_t, FILE*)' declared with attribute 'warn_unused_result' []
   51 |     fread(&c, sizeof(char), 1, fp);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
Src/BlockFile.cpp: In member function 'unsigned int CBlockFile::ReadDWord(uint32_t*)':
Src/BlockFile.cpp:73:8: warning: ignoring return value of 'size_t fread(void*, size_t, size_t, FILE*)' declared with attribute 'warn_unused_result' []
   73 |   fread(data, sizeof(uint32_t), 1, fp);
      |   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Generating dependencies: Src/Pkgs/unzip.c -> obj/unzip.d
Compiling              : Src/Pkgs/unzip.c -> obj/unzip.o
Generating dependencies: Src/Pkgs/ioapi.c -> obj/ioapi.d
Compiling              : Src/Pkgs/ioapi.c -> obj/ioapi.o
Generating dependencies: Src/Model3/93C46.cpp -> obj/93C46.d
Compiling              : Src/Model3/93C46.cpp -> obj/93C46.o
Generating dependencies: Src/Model3/JTAG.cpp -> obj/JTAG.d
Compiling              : Src/Model3/JTAG.cpp -> obj/JTAG.o
Src/Model3/JTAG.cpp: In member function 'void CJTAG::Write(bool, bool, bool, bool)':
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'RunTestIdle' not handled in switch []
  266 |         switch (m_state)
      |                ^
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'SelectDRScan' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'Exit1DR' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'PauseDR' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'Exit2DR' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'UpdateDR' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'SelectIRScan' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'Exit1IR' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'PauseIR' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'Exit2IR' not handled in switch []
Src/Model3/JTAG.cpp:266:16: warning: enumeration value 'UpdateIR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'TestLogicReset' not handled in switch []
  292 |         switch (m_state)
      |                ^
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'RunTestIdle' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'SelectDRScan' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'CaptureDR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'Exit1DR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'PauseDR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'Exit2DR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'SelectIRScan' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'CaptureIR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'Exit1IR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'PauseIR' not handled in switch []
Src/Model3/JTAG.cpp:292:16: warning: enumeration value 'Exit2IR' not handled in switch []
Generating dependencies: Src/Graphics/Legacy3D/Error.cpp -> obj/Error.d
Compiling              : Src/Graphics/Legacy3D/Error.cpp -> obj/Error.o
Generating dependencies: Src/Pkgs/glew.c -> obj/glew.d
Compiling              : Src/Pkgs/glew.c -> obj/glew.o
In file included from Src/Pkgs/glew.c:55:
Src/Pkgs/GL/glxew.h:98:10: fatal error: X11/Xlib.h: No such file or directory
   98 | #include <X11/Xlib.h>
      |          ^~~~~~~~~~~~
compilation terminated.
make: *** [Makefiles/Rules.inc:351: obj/glew.o] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.2, NixOS, 25.05 (Warbler), 25.05.20250412.2631b0b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.1`
 - nixpkgs: `/nix/store/5ds20jm3x2s4z7wn3581r6lc9ybmh45b-source`

### Notify maintainers

@msanft 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/su/supermodel/package.nix
+++ b/pkgs/by-name/su/supermodel/package.nix
@@ -4,25 +4,27 @@
   libGLU,
   SDL2,
   SDL2_net,
+  xorg,
   stdenv,
   zlib,
 }:
 
 stdenv.mkDerivation {
   pname = "supermodel";
-  version = "0-unstable-2024-11-07";
+  version = "0-unstable-2025-04-17";
 
   src = fetchFromGitHub {
     owner = "trzy";
     repo = "supermodel";
-    rev = "4e7356ab2c077aa3bc3d75fb6e164a1c943fe4c1";
-    hash = "sha256-ajRbgs6oMFF+dYHPsKM7FU16vuZcSovaNk2thdsUWtk=";
+    rev = "2272893a0511c0b3b50f6dda64addb7014717dd3";
+    hash = "sha256-3FdLBGxmi4Xj7ao2nvjLleJSTXvKQrhUWvnQr8DK/RY=";
   };
 
   buildInputs = [
     libGLU
     SDL2
     SDL2_net
+    xorg.libX11
     zlib
   ];
 

